packages/llvm.rb

```diff
@@ -285,51 +47,44 @@ def self.build
 cxx_inc=#{CREW_PREFIX}/include/c++/\${version}/\${machine}
 gnuc_lib=#{CREW_LIB_PREFIX}/gcc/\${machine}/\${version}
 clang++ -fPIC  -rtlib=compiler-rt -stdlib=libc++ -cxx-isystem \${cxx_sys} -I \${cxx_inc} -B \${gnuc_lib} -L \${gnuc_lib} \"\$@\"' > clc++"
-      case ARCH
-      when "x86_64"
-        system 'cmake',
-               "-DCURSES_INCLUDE_PATH='#{CREW_PREFIX}/include/ncursesw'",
-               "-DCMAKE_INSTALL_PREFIX=#{CREW_PREFIX}",
-               '-DCMAKE_CXX_FLAGS="-fPIC"',
-               '-DCMAKE_BUILD_TYPE=Release',
-               '-DLLVM_LIBDIR_SUFFIX=64',
-               '-DBUILD_SHARED_LIBS=ON',
-               '-DLLVM_ENABLE_RTTI=ON',
-               '-Wno-dev',
-               '..'
-               # Fix for file INSTALL cannot find ".../lib64/python2.7" error.
-               # See http://lists.llvm.org/pipermail/lldb-dev/2015-June/007633.html.
-               system "sed -i '40,43d' tools/lldb/cmake_install.cmake"
-      else # armv7l, aarch64 or i686
-        system 'cmake',
-               "-DCURSES_INCLUDE_PATH='#{CREW_PREFIX}/include/ncursesw'",
-               "-DCMAKE_INSTALL_PREFIX=#{CREW_PREFIX}",
-               '-DCMAKE_CXX_FLAGS="-fPIC"',
-               '-DCMAKE_BUILD_TYPE=Release',
-               '-DBUILD_SHARED_LIBS=ON',
-               '-DLLVM_ENABLE_RTTI=ON',
-               '-Wno-dev',
-               '..'
-      end
-      system 'make'
+
+        system "cmake -G Ninja \
+            -DCMAKE_INSTALL_PREFIX=#{CREW_PREFIX} \
+            -DLLVM_DEFAULT_TARGET_TRIPLE=#{ARCH}-cros-linux-gnu \
+            -DLLVM_TARGETS_TO_BUILD=#{LLVM_TARGETS_TO_BUILD} \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DLLVM_LIBDIR_SUFFIX='#{CREW_LIB_SUFFIX}' \
+            -DLLVM_BUILD_LLVM_DYLIB=ON \
+            -DLLVM_ENABLE_RTTI=ON \
+            -DLLVM_USE_LINKER=gold \
+            -DLLVM_INSTALL_UTILS=ON \
+            -DLLVM_BINUTILS_INCDIR='../binutils/include' \
+            -DLLVM_ENABLE_PROJECTS='clang;clang-tools-extra;libcxx;libcxxabi;libunwind;lldb;compiler-rt;lld;polly;openmp' \
```

Comment:
So this replaces all the long build steps above?  That would an awesome change.

Reply:
Probably due to build setup changes over the last several years... and using ninja is also a huge plus.

Reply:
My current biggest annoyance though is that there's no reason (I think) that we need to enable ALL llvm targets, aside from this issue I'm seeing with llvm components not otherwise being listed as built via ```llvm-config``` - which otherwise causes mesa's config to fail. Just building the host-specific targets would make this package much smaller.

Of course, this is fixed upstream in the master branch but not in the llvm-11 branch.